src/udev/udevd.c

```diff
@@ -305,6 +305,7 @@ static void manager_exit(Manager *manager) {
                   "STATUS=Starting shutdown...");
 
         /* close sources of new events and discard buffered events */
+        udev_ctrl_cleanup(manager->ctrl);
```

Comment:
I think it is better to call `udev_ctrl_cleanup()` in `udev_ctrl_free()`.

Reply:
How could I differentiate between a client process (udevadm) and the daemon/manager (udevd) from within `udev_ctrl_free()`? We don't want clients to unlink the control socket.

Reply:
Oh, I suppose `UdevCtrl.cleanup_socket` would be false in the case of a client connection.

Reply:
If we move it to udev_ctrl_free(), is it possible that udevd child processes might unlink the socket?

Reply:
> If we move it to udev_ctrl_free(), is it possible that udevd child processes might unlink the socket?

This is exactly what happens.

Reply:
It looks like my patch has the same flaw. Will revise.